direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/BinaryBlobFormat.java

```diff
@@ -116,22 +116,29 @@ public void close() throws IOException {
         blobStream = null;
       }
     }
+
+    @Override
+    public Path getPath() {
+      return path;
+    }
   }
 
-  public static class Reader implements Iterable<StreamElement>, Closeable {
+  public static class BinaryBlobReader implements Reader {
 
+    private final Path path;
     private final Parser<Serialization.Element> parser = Serialization.Element.parser();
     private final EntityDescriptor entity;
     private final Serialization.Header header;
     private final String blobName;
     private DataInputStream blobStream = null;
```

Comment:
final

Reply:
That's not possible, because `close` method should be idempotent and must therefore set this to `null`.